go-controller/pkg/ovn/egressip.go

```diff
@@ -466,14 +466,19 @@ func (e *egressIPMode) createEgressPolicy(podIps []net.IP, status egressipv1.Egr
 }
 
 func (e *egressIPMode) deleteEgressPolicy(podIps []net.IP, status egressipv1.EgressIPStatusItem, egressIPName string) error {
+	isEgressIPv6 := utilnet.IsIPv6String(status.EgressIP)
+	gatewayRouterIP, err := e.getGatewayRouterJoinIP(status.Node, isEgressIPv6)
+	if err != nil {
+		return fmt.Errorf("unable to retrieve gateway IP for node: %s, err: %v", status.Node, err)
+	}
 	for _, podIP := range podIps {
 		var filterOption string
 		if utilnet.IsIPv6(podIP) && utilnet.IsIPv6String(status.EgressIP) {
 			filterOption = fmt.Sprintf("ip6.src == %s", podIP.String())
 		} else if !utilnet.IsIPv6(podIP) && !utilnet.IsIPv6String(status.EgressIP) {
 			filterOption = fmt.Sprintf("ip4.src == %s", podIP.String())
 		}
-		policyIDs, err := findLRPolicyIDs(filterOption, egressIPName)
+		policyIDs, err := findLRPolicyIDs(filterOption, egressIPName, gatewayRouterIP)
```

Comment:
As discussed today, this is currently a problem. I am writing this up for future reference, you can ignore it Tim, as you're already aware. 

So, today the egress IP functionality is supposed to support HA mode, that being: if a node falls off the radar, any secondary egress IP will continue to be used and render the functionality still functional. This is the background for having multiple router policies. We can thus have pod A running on node X with egress IP assignments on node Y and node Z, if node Z goes down at any moment: node Y still continues serving that egress IP. However, as you've noticed this is not true. The second logical router policy applied in openflow rules on node Z will overwrite the first one created for node Y. Hence if node Z goes down _silently_, egress IP is broken.  

Now, I mentioned _silently_ above because the code does take into account a node falling off the cluster in kubernetes eyes. If the node disappears from the API, we will remove the logical router policy for that node and hence continue serving traffic through node Y. 

What is thus needed to solve this problem is to use ECMP routes combined with BFD. Two issue with that though: OVN does not support ECMP routes for "logical router policies" only for "logical router static routes", which we cannot use because that would break east <-> west traffic for any egress pod. And secondly, OVN's ECMP solution does not support BFD, which is essential for this use case, though ECMP in general would help. 

Anyhow, given that there's an RFE to use BFD for ECMP, I will also probably create an RFE to OVN so that they implement ECMP for logical router policies. 

For now, a user could delete any cluster node that has detected broken networking. As such: this problem is worked-around.